src/traces/scatterternary/defaults.js

```diff
@@ -0,0 +1,89 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+
+var constants = require('../scatter/constants');
+var subTypes = require('../scatter/subtypes');
+var handleMarkerDefaults = require('../scatter/marker_defaults');
+var handleLineDefaults = require('../scatter/line_defaults');
+var handleLineShapeDefaults = require('../scatter/line_shape_defaults');
+var handleTextDefaults = require('../scatter/text_defaults');
+var handleFillColorDefaults = require('../scatter/fillcolor_defaults');
+
+var attributes = require('./attributes');
+
+
+module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
+    }
+
+    var a = coerce('a') || [],
```

Comment:
Actually, thanks for pointing this out, we shouldn't treat `[]` as equivalent to `undefined`, I'll change it. We already don't treat them the same in other trace types (at least scatter), and for good reason:

If the arrays have different length, we take the minimum and throw away the extra values from the longer arrays. So why should length zero be different? Lets say a developer already knows the x data but the y is yet to be measured; they can set up the trace with the x data and an empty y array, and just fill in the y values as they arrive. If `[]` were the same as `undefined`, then the trace would revert to `y0` and `dy` mode at the beginning, which defaults to filling in integers from zero. Not at all what they wanted! The same applies to scatterternary, since it can fill in one missing array with the `sum` minus the other two. I have no idea if chemists would ever really operate this way, but I could imagine the same, they know the quantities of `a` and `b` they're going to use ahead of time, then they add `c` to saturation or something.